proof_of_play_test.go

```diff
@@ -26,3 +34,47 @@ func TestStop(t *testing.T) {
 
 	pop.Expire(ad)
 }
+
+func TestIsLeaseExpired(t *testing.T) {
+	pop := NewProofOfPlay(nil)
+	pop.Stop()
+
+	expired := float64(time.Now().Add(-1 * time.Hour).Unix())
+	ad := Ad{"lease_expiry": expired}
+	isExpired := pop.isLeaseExpired(ad)
+	assert.True(t, isExpired)
+
+	unexpired := float64(time.Now().Add(1 * time.Hour).Unix())
+	ad2 := Ad{"lease_expiry": unexpired}
+	isExpired = pop.isLeaseExpired(ad2)
+	assert.False(t, isExpired)
+}
+
+func TestRetryPoP(t *testing.T) {
+	pop := &proofOfPlay{
+		requests:   make(chan *PoPRequest, 100),
+		retryQueue: make(chan *PoPRequest, 100),
+	}
+
+	now := time.Now()
+	reqTime := now.Add(-59000 * time.Millisecond)
+	unexp := float64(now.Add(1 * time.Hour).Unix())
+	unexpAd := Ad{"id": "unexp", "lease_expiry": unexp}
+	unexpReq := &PoPRequest{Ad: unexpAd, Status: true, RequestTime: reqTime}
+	pop.retryPoP(unexpReq)
+	assert.Equal(t, len(pop.requests), 1)
+
+	exp := float64(now.Add(-1 * time.Hour).Unix())
+	expiredAd := Ad{"id": "expired", "lease_expiry": exp}
+	expiredReq := &PoPRequest{Ad: expiredAd, Status: true, RequestTime: reqTime}
+	pop.retryPoP(expiredReq)
+	assert.Equal(t, len(pop.requests), 1)
+
+	unexpReq = &PoPRequest{Ad: unexpAd, Status: false, RequestTime: reqTime}
+	pop.retryPoP(unexpReq)
+	assert.Equal(t, len(pop.requests), 2)
+
+	end := time.Now()
+	assert.True(t, now.Before(end))
+	pop.Stop()
```

Comment:
I'd move this to a defer statement after you create pop. Looks like you just want to make sure the channels are closed and goroutines are not hanging after the test.